test/CompressedRelationsTest.cpp

```diff
@@ -504,6 +510,57 @@ TEST(CompressedRelationWriter, getFirstAndLastTriple) {
                         ::testing::Eq(std::nullopt));
 }
 
+// _____________________________________________________________________________
+TEST(CompressedRelationWriter, getFirstAndLastTripleWithUpdates) {
+  // A dummy graph ID.
+
+  // Set up a permutation with three triple: (1, 2, 3) (1, 3, 4) (1, 4, 5), all
+  // in the same graph. Each triple will be stored in its own block, to make the
+  // manual playing with locatedTriples easier.
+  int g2 = 120349;
+  std::vector<RelationInput> inputs;
+  inputs.push_back(RelationInput{1, {{2, 3, g2}, {3, 4, g2}, {4, 5, g2}}});
+
+  auto filename = "getFirstAndLastTriple2.dat";
+  auto [blocks, metaData, readerPtr] =
+      writeAndOpenRelations(inputs, filename, 0_B);
+
+  // Set up located triples that delete the first triple. This has the
+  // consequence that the first triple of the relation `1` actually lies in the
+  // second block of that relation, because the first block is completely empty
+  // after the update.
+  LocatedTriplesPerBlock locatedTriples;
+  std::vector<LocatedTriple> deleteTriples;
+  deleteTriples.emplace_back(
+      LocatedTriple{0, IdTriple{{V(1), V(2), V(3), V(g2)}}, false});
+  locatedTriples.setOriginalMetadata(blocks);
+  locatedTriples.add(deleteTriples);
+
+  // Test infrastructure.
+  auto testFirstAndLastBlock = [&](ScanSpecification spec, auto matcher) {
```

Comment:
Please add the `ad_utility::source_location` + `generateLocationTrace`, so any failures could be traced to the actual invocation in the future.